src/App.jsx

```diff
@@ -1,7 +1,36 @@
+import { useState, useEffect } from 'react';
+import jsonData from './data.json';
+import './App.css';
+
 const App = () => {
```

Comment:
App.jsx should only maintain components itself, good practive not to make api call in App.jsx
It is not clear what u do at the bottom and what it stands for 

please transfer all the logic to corresponding component